### ISSUE TITLE
Command API for changing focus from the neovim iframe

### ISSUE THREAD

**Issue by infokiller** [glacambre/firenvim#236]
Thanks for this great project!
After I start editing a textarea in the neovim iframe, is there a way to focus back on the original page using a keybinding?
I can do this by clicking with the mouse outside the iframe.

Thanks!

**Comment by glacambre** [comment]
There's no way to do that for now but this is something I've been thinking about for a while, so it'll happen some day :)

**Comment by alerque** [comment]
This would be quite handy, especially when using keyboard driven browser interfaces.

Right now the only way I know to do it is `:wq` out of the editor, then tab away from the textbox input. That restores keyboard focus to the page the way it would normally be. Beware it does leave something selected, frequently the "submit" button. <kbd>Shift</kbd><kbd>Tab</kbd> might be safer in that respect.

**Comment by glacambre** [comment]
You can now give focus back to the page by calling `firenvim#focus_page()`. I'm keeping this issue open until this is also possible to do with the [command API](https://github.com/glacambre/firenvim/pull/240#issuecomment-554774036).

**Comment by infokiller** [comment]
Thanks @glacambre I tried this but it doesn't work for me in Chromium. I also tried something like
```javascript
setTimeout( () => { console.log('now'); document.documentElement.focus(); } , 5000)
```
from the console and then focusing on the neovim iframe, and it didn't work too.

**Comment by glacambre** [comment]
@infokiller fixed in #242

**Comment by infokiller** [comment]
Thanks! works indeed!

**Comment by glacambre** [comment]
Reopening because we still need a command-api versioon

**Comment by infokiller** [comment]
Sorry for closing, that was accidental /:

**Comment by glacambre** [comment]
No problem :)

**Comment by infokiller** [comment]
I played a bit with the focus change keybinding. Overall it's definitely an improvement, but there's still an issue for me. 
Typically, after I navigate to a webpage and want to edit a text area, I will use a keybinding defined in a browser navigation extension (currently using vimium and the keybinding is `gi`). This keybinding allows me to navigate between the input fields in the page and select the one I want to edit.
The issue in this workflow is that once the first textarea field is focused, firenvim creates the iframe and steals the focus, so I can no longer keep typing `gi` to navigate to the next input field. If I use the keybinding to change focus from the neovim frame, I can use the navigation commands in vimium again, but usually it will just change focus to the first input field again. Leaving aside the issue that then I can't navigate to another input field, after the first input field is focused again, the firenvim iframe seems to get focus, but it actually doesn't capture my keys. I can work around that by pressing `Tab` and then `Shift-Tab`.

I'm not sure how much of this issue is related to vimium, so I'll test it with different browser navigation extensions.
It would be cool to have an enhanced "unfocus firenvim" that actually iterates over all relevant input fields in the page (like vimium does with `gi`), and focuses back on the page when it wraps around. 
For example, say we have two textarea elements in the page: `textarea1` and `textarea2`. When the hits this new keybinding, firenvim opens the iframe for `textarea1` and focuses it. Then, if the use hits the keybinding again, a new firenvim iframe for `textarea2` will be created and focused. If the user hits the keybinding once again, firenvim brings focus back to the page.

**Comment by glacambre** [comment]
There's nothing I can do about the navigation issue - when an input element is focused, it must get a neovim frame. What you could do is do the same thing as https://github.com/glacambre/firenvim/issues/203: only manually trigger the firenvim frame.

> after the first input field is focused again, the firenvim iframe seems to get focus, but it actually doesn't capture my keys

This I should fix.

>It would be cool to have an enhanced "unfocus firenvim" that actually iterates over all relevant input fields in the page (like vimium does with gi), and focuses back on the page when it wraps around.

This is not something Firenvim should handle, navigation is another extension's job. What I could do though is implement a `firenvim#focus_input()` function that gives focus back to the input field rather than to the page.

**Comment by infokiller** [comment]
> There's nothing I can do about the navigation issue - when an input element is focused, it must get a neovim frame. What you could do is do the same thing as #203: only manually trigger the firenvim frame.
> 
> > after the first input field is focused again, the firenvim iframe seems to get focus, but it actually doesn't capture my keys
> 
> This I should fix.
> 
> > It would be cool to have an enhanced "unfocus firenvim" that actually iterates over all relevant input fields in the page (like vimium does with gi), and focuses back on the page when it wraps around.
> 
> This is not something Firenvim should handle, navigation is another extension's job. What I could do though is implement a `firenvim#focus_input()` function that gives focus back to the input field rather than to the page.

I agree that it seems out of scope for Firenvim. I'm thinking how I could implement this, externally to the extension. I'll see if I can get it to work using a separate extension.

**Comment by glacambre** [comment]
Closed in #364. I'll probably publish a release soon.
